accumulator/script.js

```diff
@@ -0,0 +1,710 @@
+                },
+                {
+                    formula_id: 'cell_energy_and_acc_n_series_and_acc_n_par',
+                    necessary_ids: ['acc_n_series', 'acc_n_par', 'cell_energy'],
+                    calculate: (segs_in_par, acc_n_series, acc_n_par, cell_energy) => acc_n_series * acc_n_par * cell_energy
+                },
+                {
+                    formula_id: 'acc_capacity_and_acc_max_voltage',
+                    necessary_ids: ['acc_capacity', 'acc_max_voltage'],
+                    calculate: (segs_in_par, acc_capacity, acc_max_voltage) => acc_capacity * acc_max_voltage
+                }
+            ]
+        },
+        {
+            id: 'acc_equivalent_resistance',
+            placeholder: 'Accumulator Equivalent Resistance',
+            inputValue: null,
+            formulas: [
+                {
+                    formula_id: 'cell_internal_resistance_and_acc_n_series_and_acc_n_par',
+                    necessary_ids: ['cell_internal_resistance', 'acc_n_series', 'acc_n_par'],
+                    calculate: (segs_in_par, cell_internal_resistance, acc_n_series, acc_n_par) => cell_internal_resistance * acc_n_series / acc_n_par
+                },
+                {
+                    formula_id: 'n_seg_and_seg_internal_resistance',
+                    necessary_ids: ['n_seg', 'seg_equivalent_resistance'],
+                    calculate: (segs_in_par, n_seg, seg_equivalent_resistance) => !segs_in_par ? seg_equivalent_resistance * n_seg : seg_equivalent_resistance / n_seg
+                }
+            ]
+        },
+        {
+            id: 'accumulator_maximum_current',
+            placeholder: 'Accumulator Maximum Current',
+            inputValue: null,
+            formulas: [
+                {
+                    formula_id: 'acc_max_voltage, acc_equivalent_resistance and acc_power',
+                    necessary_ids: ['acc_max_voltage', 'acc_equivalent_resistance', 'acc_power'],
+                    calculate: (segs_in_par, acc_max_voltage, acc_equivalent_resistance, acc_power) => (acc_max_voltage + Math.sqrt(acc_max_voltage**2 - 4 * acc_equivalent_resistance * acc_power)) / (2 * acc_equivalent_resistance)
+                }
+            ]
+        },
+        {
+            id: 'accumulator_maximum_current_2',
+            placeholder: 'Accumulator Maximum Current (Second Value)',
+            inputValue: null,
+            formulas: [
+                {
+                    formula_id: 'acc_max_voltage, acc_equivalent_resistance and acc_power',
+                    necessary_ids: ['acc_max_voltage', 'acc_equivalent_resistance', 'acc_power'],
+                    calculate: (segs_in_par, acc_max_voltage, acc_equivalent_resistance, acc_power) => (acc_max_voltage - Math.sqrt(acc_max_voltage**2 - 4 * acc_equivalent_resistance * acc_power)) / (2 * acc_equivalent_resistance)
+                }
+            ]
+        },
```

Comment:
These formulas are getting me wrong values

Reply:
Can you give me the scenario